xdsl/frontend/onnx/context.py

```diff
@@ -0,0 +1,76 @@
+from onnx import GraphProto, NodeProto, ValueInfoProto
+
+from xdsl.builder import ImplicitBuilder
+from xdsl.dialects import func, onnx
+from xdsl.dialects.builtin import ModuleOp
+from xdsl.frontend.onnx.type import get_type
+from xdsl.ir import Attribute, SSAValue
+from xdsl.irdl import IRDLOperation
+
+OP_BY_OP_TYPE: dict[str, type[IRDLOperation]] = {
+    "Add": onnx.Add,
+    "Sub": onnx.Sub,
+}
+"Dictionary that associates the name of the operations with the respective operation in ONNX dialect."
+
+
+class Ctx:
+    "Class to represent the Context."
+    type_by_name: dict[str, Attribute]
+    value_by_name: dict[str, SSAValue]
+
+    def __init__(self):
+        "Constructor to initialize the Context class."
+        self.type_by_name = {}
+        self.value_by_name = {}
+
+
+def visit_node(node: NodeProto, ctx: Ctx) -> None:
```

Comment:
Can you please add documentation for how this function and the functions below work?